rakelib/artifacts.rake

```diff
@@ -54,20 +54,44 @@ namespace "artifact" do
     File.open(".bundle/config", "w") { }
   end
 
-  # locate the "gem "logstash-core" ..." line in Gemfile, and if the :path => "." option if specified
+  # locate the "gem "logstash-core" ..." line in Gemfile, and if the :path => "..." option if specified
   # build and install the local logstash-core gem otherwise just do nothing, bundler will deal with it.
   task "install-logstash-core" do
+    RE = /^\s*gem\s+["']logstash-core["'](?:\s*,\s*["'][^"^']+["'])?(?:\s*,\s*:path\s*=>\s*["']([^"^']+)["'])?/i
+
     lines = File.readlines("Gemfile")
-    matches = lines.select{|line| line[/^gem\s+["']logstash-core["']/i]}
+    matches = lines.select{|line| line[RE]}
```

Comment:
no, the current code also catches multiple matches which we would loose by using find and != nil